src/gateways/ElasticsearchGateway.test.ts

```diff
@@ -0,0 +1,47 @@
+import elasticsearch from '@elastic/elasticsearch';
+import { NoOpLogger } from '../logging/NoOpLogger';
+import { ElasticsearchGateway } from "./ElasticsearchGateway";
+
+describe('ElasticsearchGateway', () => {
+  const indexName = 'documents';
+  const metadata = { documentId: 'gs523ad', ey: 'value' };
+
+  let client: elasticsearch.Client;
+  let gateway: ElasticsearchGateway;
+
+  beforeEach(() => {
+    client = {
+      index: jest.fn(() => Promise.resolve())
+    } as unknown as elasticsearch.Client;
```

Comment:
Haha - agree this is madness!

It's because `client` is missing a bunch of other stuff from `elasticsearch.Client` (all the other methods) so I can't cast it directly to `elasticsearch.Client` because TS is trying to be too clever and says "you can't cast this because it's nothing like what you're trying to cast to". So if we cast to an `unknown` first it gets us around this issue.

Reply:
Could probably just turn off some of the TypeScript checking in test files, but it's kinda useful at the same time.

Reply:
Haha all good!